src/Shared/ThrottlingDomainEventBehavior.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading.RateLimiting;
+using Microsoft.Extensions.Options;
+
+namespace Workleap.DomainEventPropagation;
+
+internal class ThrottlingDomainEventBehavior(IOptions<EventPropagationThrottlingOptions> options) : ISubscriptionDomainEventBehavior, IDisposable
+{
+    private readonly RateLimiter _rateLimiter = new TokenBucketRateLimiter(new TokenBucketRateLimiterOptions
+    {
+        TokenLimit = options.Value.MaxEventsPerSecond,
+        QueueProcessingOrder = QueueProcessingOrder.OldestFirst,
+        QueueLimit = options.Value.QueueLimit,
+        ReplenishmentPeriod = TimeSpan.FromSeconds(1),
+        TokensPerPeriod = options.Value.MaxEventsPerSecond,
+        AutoReplenishment = true
+    });
+
+    public async Task HandleAsync(DomainEventWrapper domainEvent, DomainEventHandlerDelegate next, CancellationToken cancellationToken)
+    {
+        using var lease = await this._rateLimiter.AcquireAsync(1, cancellationToken).ConfigureAwait(false);
+
+        if (lease.IsAcquired)
+        {
+            await next(domainEvent, cancellationToken).ConfigureAwait(false);
+        }
+        else
+        {
+            throw new ThrottlingException();
```

Comment:
Do we want to provide more context here, like the type of event being throttled? I believe this `ThrottlingException` is an internal exception that isn't supposed to bubble up to the consumer side and should be caught in order to return HTTP status code 429?